syncer/etcd/agent.go

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package etcd
+
+import (
+	"github.com/apache/servicecomb-service-center/pkg/log"
+	"github.com/apache/servicecomb-service-center/syncer/servicecenter"
+	"github.com/coreos/etcd/embed"
+	"github.com/coreos/etcd/etcdserver/api/v3client"
+)
+
+// Agent warps the embed etcd
+type Agent struct {
+	conf    *Config
+	etcd    *embed.Etcd
+	storage servicecenter.Storage
+}
+
+// NewAgent new etcd agent with config
+func NewAgent(conf *Config) *Agent {
+	return &Agent{conf: conf}
+}
+
+// Run etcd agent
+func (a *Agent) Run() error {
+	etcd, err := embed.StartEtcd(a.conf.Config)
+	if err != nil {
+		return err
+	}
+	select {
+	case <-etcd.Server.ReadyNotify():
+		log.Info("ready notify")
+	case <-etcd.Server.StopNotify():
```

Comment:
Could you tell us what situation will receive from channel of `etcd.Server.StopNotify()`?
Suggest to give some comment here.

Reply:
"StopNotify returns a channel that receives an empty struct When the server is stopped", That is the introduction of etcd about it. However, if an error occurs at runtime, etcd will send a notification to errorch, we can get it from "etcd.Servcer.Err()".

Reply:
I think  although this is a rare occurrence, syncer should capture it, suggest to return an err like `etcd.Servcer.Err()` when etcd return through the `StopNotify`.